libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -137,22 +153,36 @@ internal class RouteAlternativesController constructor(
         observerProcessingJob?.cancel()
     }
 
+    fun getMetadataFor(navigationRoute: NavigationRoute): AlternativeRouteMetadata? {
+        return metadataMap[navigationRoute.id]
+    }
+
     private val nativeObserver = object : com.mapbox.navigator.RouteAlternativesObserver {
         override fun onRouteAlternativesChanged(
             routeAlternatives: List<RouteAlternative>,
             removed: List<RouteAlternative>
         ): List<Int> {
-            val routeProgress = tripSession.getRouteProgress()
-                ?: return emptyList()
+            logD("${routeAlternatives.size} native alternatives available", LOG_CATEGORY)
+            if (paused) {
+                logD("paused, returning", LOG_CATEGORY)
+                return emptyList()
+            }
```

Comment:
How can this happen? As I see NN tracks current route without our help. Why do they trigger alternatives update if they're already processing a new route?

Reply:
Do I understand correctly that:
1. Android SDK gets alternatives form NN in `onRouteAlternatives`
2. Android SDK sets alternatives back to NN using `setAlternativeRoutes`
3. NN triggers `onRouteAlternatives` with the same alternatives the SDK set in step 2
?

Reply:
Yes, but how we obtained alternatives is irrelevant, the problem is that (before the changes in this PR):
1. Developers set routes to Nav SDK.
2. Nav SDK calls `RoutesObserver`.
3. Nav Native processes the routes and in the middle of processing calls `onRouteAlternatives` _with the same routes developer just set and received in `RoutesObserver`_.
4. Developer has a logic in `onRouteAlternatives` to update the routes and set them back to Nav SDK, creating an unnecessary, duplicate update.
6. `onRouteAlternatives` would not fire a second time with the same alternative so there's no endless loop, but there is one unnecessary callback.